onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -0,0 +1,287 @@
+        if not update:
+            if not set(fields).issubset(request_data.keys()):
+                return results(
+                    error=True,
+                    message="Fields object_id, clazz and name are required.",
+                    data=None
+                )
+
+        fields.append('active')
+
+        # check if invalid fields are provided
+        if any(a not in fields for a in request_data.keys()):
+            return results(
+                error=True,
+                message="Valid fields are object_id, clazz and name.",
+                data=None
+            )
+
+        data = {}
+        for key in fields:
+            available = request_data.get(key) is not None
+            available and data.update({key: request_data.get(key)})
+
+        return results(error=False, message=None, data=data)
+
+    def get_tableau_type(self, xform_type):
+        '''
+        Returns a tableau-supported type based on a xform type.
+        '''
+        tableau_types = {
+            'integer': 'int',
+            'decimal': 'float',
+            'dateTime': 'datetime',
+            'text': 'string'
+        }
+
+        return tableau_types.get(xform_type, 'string')
+
+    def flatten_xform_columns(self, json_of_columns_fields):
+        '''
+        Flattens a json of column fields and the result is set to a class
+        variable.
+        '''
+        for a in json_of_columns_fields:
+            self.flattened_dict[a.get('name')] = self.get_tableau_type(
+                a.get('type')
+            )
+            # using IGNORED_FIELD_TYPES so that choice values are not included.
+            if a.get('children') and a.get('type') not in IGNORED_FIELD_TYPES:
+                self.flatten_xform_columns(a.get('children'))
+
+    def get_tableau_column_headers(self):
+        '''
+        Retrieve columns headers that are valid in tableau.
+        '''
+        tableau_colulmn_headers = []
+
+        def append_to_tableau_colulmn_headers(header, question_type=None):
+            quest_type = 'string'
+            if question_type:
+                quest_type = question_type
+
+            # alias can be updated in the future to question labels
+            tableau_colulmn_headers.append({
+                'id': header,
+                'dataType': quest_type,
+                'alias': header
+            })
+
+        # using nested loops to determine what valid data types to set for
+        # tableau.
+        for header in self.xform_headers:
+            for quest_name, quest_type in self.flattened_dict.items():
+                if header == quest_name or header.endswith('_%s' % quest_name):
+                    append_to_tableau_colulmn_headers(header, quest_type)
+                    break
+            else:
+                append_to_tableau_colulmn_headers(header)
+
+        return tableau_colulmn_headers
+
+    def create(self, request, *args, **kwargs):
+        if request.user.is_anonymous():
```

Comment:
Done. I have set the value of `permission_classes` to `(OpenDataViewSetPermissions,)` - `OpenDataViewSetPermissions` is the new permissions class.